### DISCUSSION TITLE
User Registration

### DISCUSSION THREAD

**Discussion by Mertcikla** [UniTime/unitime#163] [Ideas]
User registration (especially student) is needed for our use case where our automation software does not interface with UniTime.
UniTime letting users self register with captcha and password reminder functionality would help us  #immensely.

**Comment by tomas-muller** [comment]
There are no plans to add this functionality at the moment. Especially for students, there is typically more information required than just an email (university issued id, curriculum information, etc.). Do you really which for any user to start requesting courses and/or enrolling in the classes?

Please note that when an [LDAP](http://help.unitime.org/LDAP) or a [CAS](http://help.unitime.org/CAS) authentication is enabled, no additional registration in UniTime is needed for an authenticated user to see or request events, view class or examination schedule etc. (depends on the No Role user role permissions and the status of the academic session).

**Comment by Mertcikla** [comment]
Ideally we would limit registration to university emails by domain but it is not necessary. 

Since our automation software does not support either LDAP or CAS. We are currently entering each curriculum as a hard constraint and ignoring all of the students requests to take courses out of their curriculum. User registration would enable us improve that, see exact course demands and let UniTime handle it.
